merlin/server/server_setup.py

```diff
@@ -0,0 +1,291 @@
+"""Main functions for instantiating and running Merlin server containers."""
+
+import enum
+import logging
+import os
+import shutil
+import socket
+import subprocess
+import time
+
+from merlin.server.server_config import (
+    MERLIN_CONFIG_DIR,
+    MERLIN_SERVER_CONFIG,
+    MERLIN_SERVER_SUBDIR,
+    dump_process_file,
+    parse_redis_output,
+    pull_process_file,
+    pull_server_config,
+)
+
+
+# Default values for configuration
+CONFIG_DIR = "./merlin_server/"
+IMAGE_NAME = "redis_latest.sif"
+PROCESS_FILE = "merlin_server.pf"
+CONFIG_FILE = "redis.conf"
+REDIS_URL = "docker://redis"
+CONTAINER_TYPES = ["singularity", "docker", "podman"]
+
+LOG = logging.getLogger("merlin")
+
+
+class ServerStatus(enum.Enum):
+    """
+    Different states in which the server can be in.
+    """
+
+    NOT_INITALIZED = 0
```

Comment:
Maybe have RUNNING as state 0, consistent with the notion that 0 is usually no-error. The others seem like errors. (I know their Eums and not really integers, so maybe it doesn't matter). Thoughts @koning ?

Reply:
Yes, nonzero is usually a error. RUNNING as state 0 is good with the default state set as NOT_RUNNING.

Reply:
Changed the RUNNING state as 0